File: NationalSecurityCommunity/app/src/main/java/com/national/security/community/ui/main/MainContract.java
package com.national.security.community.ui.main;

import com.national.security.community.mvp.CommonView;
import com.national.security.community.mvp.IPresenter;

public class MainContract {

    public interface View extends CommonView {
        void granted();

        void denied_never_ask_again();

        void denied();
    }

    interface Presenter extends IPresenter<View> {
        void loadData(String url);

        void requestPermission();

        void loadHome();
    }
}

File: app/src/main/java/com/jmd_news_kotlin/MainContract.kt
package com.jmd_news_kotlin

import com.jmd_news_kotlin.mvp.IPresenter
import com.jmd_news_kotlin.mvp.IView

interface MainContract {

    interface View : IView {
        fun setMsg()
    }

    interface Presenter : IPresenter {
        fun getMsg()
    }
}